mindmeld/components/entity_resolver.py

```diff
@@ -42,60 +50,266 @@
+                "to use the built in embbedder for entity resolution. See "
+                "https://www.mindmeld.com/docs/userguide/getting_started.html")
+
+    def get_resolver(self, app_path, resource_loader, entity_type, **kwargs):
+        """
+        Identifies appropriate entity resolver based on input config and
+            returns it.
 
         Args:
-            app_path (str): The application path
-            resource_loader (ResourceLoader): An object which can load resources for the resolver
-            entity_type: The entity type associated with this entity resolver
-            es_host (str): The Elasticsearch host server
+            app_path (str): The application path.
+            resource_loader (ResourceLoader): An object which can load resources for the resolver.
+            entity_type (str): The entity type associated with this entity resolver.
+            er_config (dict): A classifier config
+            es_host (str): The Elasticsearch host server.
+            es_client (Elasticsearch): The Elasticsearch client.
         """
-        self._app_namespace = get_app_namespace(app_path)
-        self._resource_loader = resource_loader
-        self._normalizer = resource_loader.query_factory.normalize
+        er_config = (
+            kwargs.pop("er_config", None) or
+            get_classifier_config("entity_resolution", app_path=app_path)
+        )
+        resolver_name = er_config.get("model_type", None)
+        self._validate_resolver_name(resolver_name)
+        return ENTITY_RESOLVER_MODEL_MAPPINGS.get(resolver_name)(
+            app_path, resource_loader, entity_type, er_config, **kwargs
+        )
+
+
+class EntityResolverModelBase(ABC):
+    """
+    Base class for Entity Resolvers
+    """
+
+    def __init__(self, app_path, resource_loader, entity_type, er_config):
+        """Initializes an entity resolver"""
+        self.app_path = app_path
         self.type = entity_type
         self._is_system_entity = Entity.is_system_entity(self.type)
-        self._exact_match_mapping = None
-        self._er_config = get_classifier_config("entity_resolution", app_path=app_path)
-        self._es_host = es_host
-        self._es_config = {"client": es_client, "pid": os.getpid()}
-        self.ready = False
+        self.app_namespace = get_app_namespace(self.app_path)
+        self.name = er_config.get("model_type")
+        self.entity_map = resource_loader.get_entity_map(self.type)
+        self.normalizer = resource_loader.query_factory.normalize
 
-        if self._is_system_entity:
-            canonical_entities = []
-        else:
-            canonical_entities = self._resource_loader.get_entity_map(self.type).get(
-                "entities", []
-            )
-        self._no_canonical_entity_map = len(canonical_entities) == 0
+        # default configs useful for obtaining model cache path
+        default_er_config = DEFAULT_ENTITY_RESOLVER_MODEL_CONFIGS[self.name]
```

Comment:
`default_er_config = copy.deepcopy(DEFAULT_ENTITY_RESOLVER_MODEL_CONFIGS[self.name])`

else, the subsequent lines with change the contents of `DEFAULT_ENTITY_RESOLVER_MODEL_CONFIGS `, which will cause issues if some other line referencing it gets executed

Reply:
With new way of getting defaults, this variable won't exist anymore. So ignoring.